tests/test_callprofiler.py

```diff
@@ -0,0 +1,185 @@
+            self.fail(f"JSON data is invalid: {e.message}")
+
+    def test_single_function_profiling(self):
+
+        @profile_function
+        def foo():
+            busy_loop(0.1)
+
+        modmesh.call_profiler.reset()
+        foo()
+        root_result = modmesh.call_profiler.result()
+        self.assertEqual(len(root_result["children"]), 1)
+        self.assertEqual(root_result["children"][0]["name"], "foo")
+        foo_result = root_result["children"][0]
+        self.assertEqual(len(foo_result["children"]), 0)
+        self.assertEqual(foo_result["count"], 1)
+        self.assertGreaterEqual(foo_result["total_time"], 100)
+
+    def test_caller_profiling(self):
+
+        @profile_function
+        def bar():
+            busy_loop(0.5)
+
+        @profile_function
+        def foo():
+            busy_loop(0.1)
+            bar()
+
+        modmesh.call_profiler.reset()
+        foo()
+        root_result = modmesh.call_profiler.result()
+        self.assertEqual(len(root_result["children"]), 1)
+        self.assertEqual(root_result["children"][0]["name"], "foo")
+        foo_result = root_result["children"][0]
+        self.assertEqual(foo_result["count"], 1)
+        self.assertGreaterEqual(foo_result["total_time"], 600)
+
+        self.assertEqual(len(foo_result["children"]), 1)
+        self.assertEqual(foo_result["children"][0]["name"], "bar")
+        bar_result = foo_result["children"][0]
+        self.assertEqual(len(bar_result["children"]), 0)
+        self.assertEqual(bar_result["count"], 1)
+        self.assertGreaterEqual(bar_result["total_time"], 500)
+
+    def test_two_callers_profiling(self):
+
+        @profile_function
+        def bar():
+            busy_loop(0.5)
+
+        @profile_function
+        def foo():
+            busy_loop(0.1)
+            bar()
+
+        modmesh.call_profiler.reset()
+        bar()
+        foo()
+        root_result = modmesh.call_profiler.result()
+        self.assertEqual(len(root_result["children"]), 2)
+        search_bar = [i for i in root_result["children"] if i["name"] == "bar"]
+        self.assertEqual(len(search_bar), 1)
+        search_foo = [i for i in root_result["children"] if i["name"] == "foo"]
+        self.assertEqual(len(search_foo), 1)
+
+        bar_result = search_bar[0]
+        self.assertEqual(bar_result["count"], 1)
+        self.assertGreaterEqual(bar_result["total_time"], 500)
+
+        foo_result = search_foo[0]
+        self.assertEqual(foo_result["count"], 1)
+        self.assertGreaterEqual(foo_result["total_time"], 600)
+        self.assertEqual(len(foo_result["children"]), 1)
+        self.assertEqual(foo_result["children"][0]["name"], "bar")
+
+        second_bar_result = foo_result["children"][0]
+        self.assertEqual(len(second_bar_result["children"]), 0)
+        self.assertEqual(second_bar_result["count"], 1)
+        self.assertGreaterEqual(second_bar_result["total_time"], 500)
+
+    def test_get_result_during_profiling(self):
```

Comment:
As the previous unit test all assert the result after profiling a series of function calls.
But user should be able to get the profiling result within the function too, so add this test.

Reply:
Sounds good.